[KB-305] PyHat contribution (#222)
chhoumann/thesis-chemcam: report_thesis/src/sections/pyhat_contribution.tex

Reviewer: Include it somewhere?

\section{PyHAT Contribution}\label{sec:pyhat_contribution}

> As part of our research, we have made several contributions to \gls{pyhat}, which offers a user-friendly interface designed for performing machine learning and data analysis tasks specifically for hyperspectral data.

Reviewer: Implementing / contributing like this isn't really research + can simplify?
```suggestion
As part of our work, we have made several contributions to \gls{pyhat}. 
We describe these contributions here.
\gls{pyhat} offers a user-friendly interface designed for performing machine learning and data analysis tasks specifically for hyperspectral data.
```

> This collaboration was initiated through a series of discussions with two members \gls{usgs} responsible for managing \gls{pyhat}, wherein we identified mutual challenges and opportunities for integrating our solutions into the library.

Reviewer: Generally wanted to rephrase a bit here. And it isn't a library entirely. More like a tool that has a library for core functionality, including an application 'layer'.
```suggestion
Our collaboration was initiated through a series of discussions with two members from \gls{usgs} that are responsible for \gls{pyhat}, wherein we identified mutual challenges and opportunities for integrating our solutions into the tool.
```

> The largest contribution involved the integration of an automatic outlier detection method into the library's \gls{pls} implementation. 

Reviewer: ```suggestion
The largest contribution involved the integration of an automatic outlier detection method into \gls{pyhat}.
```

Reviewer: Wasn't specifically for the PLS implementation. It's a general method for outlier removal. It just uses PLS.

This method calculates the Mahalanobis distance for each data point and uses the chi-squared distribution to establish a threshold.

> Any datapoint exceeding this threshold is considered an out and removed from the dataset.

Reviewer: ```suggestion
Any datapoint exceeding this threshold is considered an outlier and removed from the dataset.
```

Reviewer: Is `out` the singular form of `outliers`?🤣

> Utilizing two intermediary \gls{pls} models, one as a reference and the other to evaluate the impact of outlier removal, the method iteratively identifies and eliminates outliers while assessing the performance of the second model against the reference model. 

Reviewer: It doesn't do that? It doesn't use two?

> If the second model demonstrates improved performance compared to the reference model, it replaces the reference model, and the process continues until no further significant improvement is detected. 

Reviewer: Oh, you're referring to the `best_or_model` variable? Yeah, that's not something we did for PyHAT. That was in our own project for logging.

> To conserve computational resources, the method halts if the error of the second model increases relative to the reference model, thus providing an early stopping mechanism.

Reviewer: Not to converse computational resources!
I can see the analogy to early stopping, but you need to provide more context for the reader to understand it.

> This contribution also included the development of a graphical user interface (GUI) component for the existing \gls{pyhat} GUI to visualize the outlier removal process in real-time.

Reviewer: ```suggestion
This contribution also included the development of a graphical user interface (GUI) component for the existing \gls{pyhat} GUI to configure and visualize the outlier removal process.
```

> This included utilities to select a threshold, select a given oxide for which to perform outlier removal, and a logging mechanism to display the number of outliers removed at each iteration in the GUI.

Reviewer: Utilities? You mean configuration options?
Logging was part of PyHAT already. We just did `print(...)` to use it.

Reviewer: I don't think the outlier removal section hits the mark. I can help you rewrite later.

> Another contribution made to \gls{pyhat} involved a fix of an important functionality in their Joint Approximation Diagonalization of Eigen-matrices (JADE) implementation.

Reviewer: jade in glossary?

Reviewer: sentence is weird as well, try rephrasing

The fix provided the ability to properly identify which of the original data points has the highest correlation with each independent component produced by JADE.
The correlation scores produced by this functionality can be used in a regression context, where a linear model learns the coefficients that best fit the relationship between the independent components and the original data points.

> Finally, contribution were made to improve the performance of various processes in \gls{pyhat}.

Reviewer: ```suggestion
Finally, we made some contributions to improve the performance of various processes in \gls{pyhat}.
```
